### PR TITLE
[FIXED JENKINS-42194] Do not display a warning when no schedules

### PR DESCRIPTION
**Pull request by stephenc**
See [JENKINS-42914](https://issues.jenkins-ci.org/browse/JENKINS-42194)

Screenshots:

Before:

ignore-post-commit-hooks-recommended

After:

empty-ok-when-ignoring

And validation is unaffected other that for the special case of ignoring post commit hooks with an empty spec:

ok-warning
validation-when-ignoring

@reviewbybees

### CODE REVIEW

**Comment by daniel-beck** [comment]
While we're fixing the warning, could we rephrase what's left to:

> No schedules so will only run due to SCM changes when triggered by a post-commit hook

And the new one should instead be:

> Post commit hooks are being ignored and no schedules so will never run due to SCM changes

**Comment by stephenc** [comment]
@daniel-beck good catch, no schedules is not really ever a warning

**Review by daniel-beck — APPROVED** [review]

File: core/src/main/java/hudson/triggers/SCMTrigger.java [hunk 1]
```diff
@@ -358,6 +360,24 @@ public FormValidation doCheckPollingThreadCount(@QueryParameter String value) {
...
+
+        /**
+         * Performs syntax check.
+         */
+        public FormValidation doCheckScmpoll_spec(@QueryParameter String value,
+                                                  @QueryParameter boolean ignorePostCommitHooks,
+                                                  @AncestorInPath Item item) {
+            if (StringUtils.isBlank(value)) {
+                if (ignorePostCommitHooks) {
+                    return FormValidation.ok(Messages.SCMTrigger_no_schedules_no_hooks());
```
Comment (jglick) [inline core/src/main/java/hudson/triggers/SCMTrigger.java]: Seems like this should be a warning—why would you even configure the trigger at all in this case?

**Review by jglick — APPROVED** [review]

File: core/src/main/java/hudson/triggers/SCMTrigger.java (hunk 1 above)
Comment (daniel-beck) [inline core/src/main/java/hudson/triggers/SCMTrigger.java]: This is how you disable post-commit hooks for a project.

Comment (jglick) [inline core/src/main/java/hudson/triggers/SCMTrigger.java]: Really? How bizarre.

Comment (stephenc) [inline core/src/main/java/hudson/triggers/SCMTrigger.java]: I know... doesn't help that you'd get a warning without this change :rolleyes:

**Pull request merged**